src/index.ts

```diff
@@ -44,11 +43,41 @@ export type MagicLinkPayload = {
   c: number
 }
 
+/**
+ * This interface declares the session data that is required for this package
+ *  to store the following details.
+ */
+type MagicLinkSessionData<User> = {
+  /**
+   * The magic link URL that was sent to the user's email.
+   */
+  magicLink: string
+  /**
+   * The email address used for authentication.
+   */
+  email: string
+  /**
+   * The authenticated user object.
+   */
+  user: User
+}
+
+/**
+ * This interface declares the session flash data that is required for this package
+ *  to store the following details.
+ */
+type MagicLinkSessionFlashData = {
+  /**
+   * The error message to display to the user.
+   */
+  error: string
+}
+
 /**
  * This interface declares what configuration the strategy needs from the
  * developer to correctly work.
  */
-export type EmailLinkStrategyOptions<User> = {
+export type EmailLinkStrategyOptions<User, SessionData, SessionFlashData> = {
```

Comment:
what do you think about an optional callback that is called right before the success redirect that allows the user to change the session? In my own app I was taking advantage of the optional successRedirect and handling the redirect myself. This allowed me to update the session with several other bits of information I wanted to assert about the verified user onto the session